File: tree.py
class Node():
    def __init__(self, data):
        self.data = data
        self.father = None
        self.children = None
        self.cost = 0

    
    def getData(self):
        return self.data
    
    def getCost(self):
        return self.cost
    
    def setCost(self, cost):
        self.cost = cost

    def getFather(self):
        return self.father

    def setFather(self, father):
        self.father = father
        father.setChildren(self)

    def getChildren(self):
        return self.children
    
    def setChildren(self, children):
        self.children= children

        if self.children!= None:
            for i in self.children:
                i.father= self

    def equals(self, node):
        return self.data == node.data
    
    def inArray(self, arr):
        for i in arr:
            if self.equals(i):
                return True
        return False

File: js/tree.js
class Node{
    constructor (data){
        this.data = data;
        this.father = null;
        this.children = null;
        this.cost = 0;
    }

    getData(){
        return this.data;
    }

    getFather(){
        return this.father;
    }

    setFather(father){
        this.fahter = father;
    }

    getChildren(){
        return this.children;
    }

    setChildren(children){
        this.children = children;

        if(children != null){
            for (let child in children){
                child.setFather(this);
            }   
        }
    }

    getCost(){
        return this.cost;
    }

    setCost(cost){
        this.cost = cost
    }

    equals(node){
        return node.getData() == this.getData();
    }

    inArray(arr){
        for (let node in arr){
            if (node.equals(this)){
                return true;
            }
        }

        return false
    }
}

export default Node